CMakeLists.txt

```diff
@@ -1,15 +1,51 @@
 cmake_minimum_required(VERSION 3.10)
-project(recycle)
+
+project(recycle
+  VERSION 6.0.0
+  DESCRIPTION "Simple resource pool for recycling resources in C++"
+  LANGUAGES CXX)
 
 # Define library
 add_library(recycle INTERFACE)
 target_compile_features(recycle INTERFACE cxx_std_14)
-target_include_directories(recycle INTERFACE src/)
+target_include_directories(recycle INTERFACE $<BUILD_INTERFACE:src> $<INSTALL_INTERFACE:include>)
```

Comment:
What is the difference between this and the original?

Reply:
The include path in the `BUILD_INTERFACE` is used during the build of this project, and the include path in the `INSTALL_INTERFACE` will be used by downstream projects that use an installed version of this project.

During the build of this project, the headers are located in `src`, but they will be installed in `<INSTALL_PREFIX>/include` (according to GNU convention).
This is why we have to differentiate and tell CMake, where to find the headers in both cases.

Reply:
I see, so if the library is being used as a CMake Package, `INSTALL_INTERFACE` will be used, but if it's being used with, e.g.,
 `add_subdirectory` or `FetchContent_Declare`, `BUILD_INTERFACE` will be used?
 Is this correctly understood?

Reply:
Exactly 👍🏻

Reply:
In our internal projects, we use `add_subdirectory` for dependencies. I just tried using these changes, and I got this error:
```
[cmake] CMake Error in resolve_symlinks/slide/CMakeLists.txt:
[cmake]   Target "recycle" contains relative path in its
[cmake]   INTERFACE_INCLUDE_DIRECTORIES:
[cmake] 
[cmake]     "src"
[cmake] 
[cmake] 
[cmake] -- Generating done
[cmake] CMake Generate step failed.  Build files cannot be regenerated correctly.
```
I search a little and found [this](https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_INCLUDE_DIRECTORIES.html), so it seems the `BUILD_INTERFACE` needs to have `${CMAKE_CURRENT_SOURCE_DIR}` prepended. At least that fixed the error.

Reply:
Hi, @globberwops; if you make this change, we should be able to merge this one finally ;-)

Cheers!

Reply:
@jpihl Done! Thanks for the reminder :)